controller/QtiCreator.php

```diff
@@ -348,4 +360,152 @@ private function validateXmlInput(string $xml)
         }
         \tao_helpers_Xml::getSimpleXml($xml);
     }
+
+    /**
+     * Download sample template for tabular import
+     *
+     * @param string 
+     * @throws common_exception_Error
+     */
+    public function downloadCsv()
+    {
+        if (!$this->hasRequestParameter('uri')) {
+            throw new common_exception_MissingParameter('uri', __METHOD__);
+        }
+
+        $metaDataArray = $this->getMetadataArray();
+
+        $headers = $this->getHeaderList();
+
+        $final_header = array_merge($headers, $metaDataArray);
+
+        $correct_response = array(
```

Comment:
Since we will have multiple types of template, how could we manage this in an extensible way? I mean, imagine that we want this example to be loaded for the current template, but a different one for another template and so on.

Reply:
It was done.